public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -161,3 +158,38 @@ export function runPartitionedQuery(datasource: LokiDatasource, request: DataQue
 
   return response;
 }
+
+function getNextRequestPointers(requests: LokiGroupedRequest, requestIndex: number, requestN: number) {
```

Comment:
i have just realized one thing... this approach alternates between logs-chunk and metric-chunk (i did the same in my draft-PR0, but .. assuming multi-request is less often happening, maybe something even simpler is enough:

i think we can also just do all-metrics-first-and-then-do-all-logs (or the opposite).

do you think that would make the code simpler?